### PR TITLE
Implement servus::Serializable interface

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
@favreau FYI

File: tests/camera.cpp [hunk 1]
```diff
@@ -123,12 +123,11 @@ BOOST_AUTO_TEST_CASE(cameraToGeneric)
     const zerobuf::render::Camera camera( zerobuf::render::Vector3f( 1, 0, 0 ),
                                           zerobuf::render::Vector3f( 0, 1, 0 ),
                                           zerobuf::render::Vector3f( 0, 0, 1 ));
-    const void* data = camera.getZerobufData();
-    const size_t size = camera.getZerobufSize();
+    const servus::Serializable::Data& zerobuf = camera.toBinary();
```
Comment (tribal-tec) [inline tests/camera.cpp]: typedef zerobuf::Data

File: tests/jsonSerialization.cpp [hunk 2]
```diff
@@ -210,12 +210,11 @@ BOOST_AUTO_TEST_CASE(zerobufFromJSON)
 BOOST_AUTO_TEST_CASE(rawZerobufToJSON)
 {
     const test::TestSchema& object( getTestObject( ));
-    const void* data = object.getZerobufData();
-    const size_t size = object.getZerobufSize();
+    const servus::Serializable::Data& zerobuf = object.toBinary();
```
Comment (tribal-tec) [inline tests/jsonSerialization.cpp]: dito

File: zerobuf/Zerobuf.cpp [hunk 3]
```diff
@@ -76,23 +76,13 @@ Zerobuf& Zerobuf::operator = ( Zerobuf&& rhs )
...
-}
-
 void Zerobuf::compact( const float threshold )
 {
     if( _allocator && getZerobufNumDynamics() > 0 )
         _allocator->compact( threshold );
 }
 
-void Zerobuf::copyZerobufData( const void* data, size_t size )
+bool Zerobuf::fromBinary( const void* data, const size_t size )
```
Comment (tribal-tec) [inline zerobuf/Zerobuf.cpp]: return false for the two last throws (logging?)

File: zerobuf/Zerobuf.h [hunk 4]
```diff
@@ -38,14 +36,29 @@ class Zerobuf
     /** Called if any data in this object is about to change. */
     virtual void notifyChanging() {}
 
-    /** Called by ZeroEQ subscriber upon receive() of this object. */
-    virtual void notifyReceived() {}
+    /** @name servus::Serializable interface. */
+    //@{
+    /** Update this serializable from its binary representation. */
+    bool fromBinary( const void* data, const size_t size ) override;
```
Comment (tribal-tec) [inline zerobuf/Zerobuf.h]: try public non-virtual, protected virtual

File: tests/camera.cpp (hunk 1 above)
Comment (eile) [inline tests/camera.cpp]: done

File: tests/jsonSerialization.cpp (hunk 2 above)
Comment (eile) [inline tests/jsonSerialization.cpp]: dito

File: zerobuf/Zerobuf.cpp (hunk 3 above)
Comment (eile) [inline zerobuf/Zerobuf.cpp]: done

File: zerobuf/Zerobuf.h (hunk 4 above)
Comment (eile) [inline zerobuf/Zerobuf.h]: done

**Pull request merged**